pkg/cmd/roachtest/cluster.go

```diff
@@ -1865,7 +1865,9 @@ func (c *clusterImpl) doDestroy(ctx context.Context, l *logger.Logger) <-chan st
 }
 
 func (c *clusterImpl) addLabels(labels map[string]string) error {
-	return roachprod.AddLabels(c.l, c.name, labels)
+	// N.B. we must sanitize the values; e.g., some test names can exceed the maximum length (63 chars in GCE).
+	// N.B. keys are expected to be sanitized already.
```

Comment:
Why is it okay to assume this?

Reply:
Mainly two reasons,

- keys are under our control (and static) whereas values are generated (and dynamic)
- sanitization can change the original value which may already be "safe"; this could break backwards compatibility

Should we reconsider and sanitize both?

Reply:
Gotcha, I think I misinterpreted `keys are expected to be sanitized already` as "we already sanitized them somewhere else" instead of "we should call `addLabels` with sanitized keys". I think it should be okay to leave as it, if someone did add an unsanitized key it should be pretty easy to spot.

Reply:
Gotcha. Lemme paraphrase this to remove the ambiguity.